examples/simple-genai-server/gameserver_npcchat.yaml

```diff
@@ -0,0 +1,61 @@
+---
+# Copyright 2024 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: agones.dev/v1
+kind: GameServer
+metadata:
+  name: gen-ai-server-npc
+spec:
+  ports:
+    - name: default
+      portPolicy: Dynamic
+      containerPort: 7654
+      protocol: TCP
+  template:
+    spec:
+      containers:
+        - name: simple-genai-game-server
+          image: us-docker.pkg.dev/agones-images/examples/simple-genai-game-server:0.1
+          # imagePullPolicy: Always  # add for development
+          env:
+            - name: GenAiEndpoint
+            # Use the service endpoint address when running in the same cluster as the inference server.
+              value: "http://npc-chat-api.genai.svc.cluster.local:80"
+            # GenAiContext is not passed to the npc-chat-api endpoint.
+            - name: GenAiNpc # False by default. Use GenAiNpc "true" when using the npc-chat-api as the GenAiEndpoint.
+              value: "true"
+            - name: SimEndpoint
+              value: "http://vertex-chat-api.genai.svc.cluster.local:80"
```

Comment:
Here use the `/genai/chat` endpoint instead. If you want to handle this as a follow-on and leave a TODO that's fine, though.